aten/src/ATen/native/mps/operations/LossOps.mm

```diff
@@ -307,6 +307,7 @@ static void nllnd_loss_backward_impl(Tensor& grad_input_arg,
     MPSGraphTensor* gradOutputTensor_ = nil;
   };
   bool isWeightsArrayValid = weight_arg.defined() && weight_arg.numel() > 0;
+  bool isTargetCasted = target_arg.scalar_type() != ScalarType::Long;
```

Comment:
This was not taken care previously because the related code complaining incompatible dtypes was enclosed by `ignore_index != -100` block.